protocols/gossipsub/src/error.rs

```diff
@@ -136,7 +137,7 @@ impl std::error::Error for ValidationError {}
 
 impl From<std::io::Error> for GossipsubHandlerError {
     fn from(error: std::io::Error) -> GossipsubHandlerError {
-        GossipsubHandlerError::Io(error)
+        GossipsubHandlerError::Codec(prost_codec::Error::from(error))
     }
 }
```

Comment:
I haven't tested but do we still need this implementation?

Reply:
I think so. I commented this out (and also just tried the original version with the Io) and protocols.rs and handler.rs both lit up:

```
the trait bound `GossipsubHandlerError: From<std::io::Error>` is not satisfied
the trait `From<prost_codec::Error>` is implemented for `GossipsubHandlerError`rustc[E0277](https://doc.rust-lang.org/error-index.html#E0277)
decoder.rs(11, 17): required by a bound in `asynchronous_codec::Decoder::Error`
```